### ISSUE TITLE
[$500] Chat - User is able to enter invalid link in "[]()" formatted link by editing the URL

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#38295]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.52-0 
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4423074
**Email or phone of affected tester (no customers):** vdargentotest+web031324@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
Pre-requisite: the user must be logged in
1. Go to any chat
2. Message1: Send the message "google.com"
3. Verify it is displayed as a formatted link
4. Message2: Send an invalid link with some random characters at the end "google.comqwert"
5. Verify it is not recognized as a valid link
6. Edit the message1 you sent and add some random characters at the end of the URL (ex: `[google.com](google.com)` changed to `[google.com](google.comqwer)`)
7. Verify it is still recognized as a valid link

## Expected Result:
The new invalid link from step 6 should not be recognized as valid after editing the valid URL

## Actual Result:
The new invalid link is recognized as valid after editing the valid URL

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/3e1f3556-19c0-4d86-9a3b-4abdadebd606

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0198d2d0a2f10a2232</li>
        <li>Upwork Job ID: 1769763511319392256</li>
        <li>Last Price Increase: 2024-03-25</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
@sonialiap  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp
CC @quinthar

**Comment by nagypite** [comment]
As much as I'd like to write a proposal, this does not seem to be an issue.

OP states:
> 6. Edit the message1 you sent and add some random characters at the end of the URL (ex: `[google.com](google.com)` changed to `[google.com](google.comqwer)`)

But in the attached video the actual change is from `[google.com](https://google.com)` to `[google.com](https://google.comqwer)`, which is correctly matched by `ExpensiMark` with `LOOSE_URL_REGEX`.

And although it seems like a bad url, it is valid in a local network setting, which seems to be the original intent behind `LOOSE_URL_REGEX` (see: https://github.com/Expensify/expensify-common/commit/0e1ae5079e93925f984068fd17e3d5e6f732e93d).

In summary: works as intended IMO.

**Comment by sonialiap** [comment]
I see what you mean. This is correct from a technical perspective but unexpected from a UX perspective. 

While it would be nice to make it work as a non-technical user might expect, I think this can be counted as an edge case. When a user clicks on the linked text with a misspelled URL then the web page shows an error saying to check the URL. I think users will be able to figure out their own way out of such a situation. Closing
